javalib/src/main/scala/java/util/IdentityHashMap.scala

```diff
@@ -0,0 +1,297 @@
+
+        def hasNext(): Boolean =
+          iter.hasNext()
+
+        def next(): K =
+          iter.next().inner
+
+        def remove(): Unit =
+          iter.remove()
+      }
+    }
+
+    def size(): Int =
+      self.size()
+
+    override def contains(o: Any): Boolean =
+      containsKey(o)
+
+    override def remove(o: Any): Boolean = {
+      val hasKey = contains(o)
+      if (hasKey) self.remove(o)
+      hasKey
+    }
+
+    override def removeAll(c: Collection[_]): Boolean =
+      if (size > c.size) {
+        c.scalaOps.foldLeft(false)((prev, elem) => this.remove(elem) || prev)
+      } else {
+        @tailrec
+        def removeAll(iter: Iterator[K], modified: Boolean): Boolean = {
+          if (iter.hasNext) {
+            if (findSame(iter.next(), c)) {
+              iter.remove()
+              removeAll(iter, true)
+            } else
+              removeAll(iter, modified)
+          } else
+            modified
+        }
+        removeAll(this.iterator, false)
+      }
+
+    override def retainAll(c: Collection[_]): Boolean = {
+      val iter = iterator()
+      var changed = false
+      while (iter.hasNext) {
+        val elem = iter.next()
+        if (!findSame(elem, c)) {
+          iter.remove()
+          changed = true
+        }
+      }
+      changed
+    }
+
+    override def clear(): Unit =
+      self.clear()
+  }
+
+  private final class EntrySet extends AbstractSet[Map.Entry[K, V]] {
+    def iterator(): Iterator[Map.Entry[K, V]] = {
+      new ju.Iterator[Map.Entry[K, V]] {
+        private val iter = inner.entrySetIterator()
+
+        def hasNext(): Boolean =
+          iter.hasNext()
+
+        def next(): Map.Entry[K, V] =
+          new MapEntry(iter.next())
+
+        def remove(): Unit =
+          iter.remove()
+      }
+    }
+
+    def size(): Int =
+      inner.size()
+
+    override def contains(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          self.containsKey(thatKey) &&
+            same(self.get(thatKey), value.getValue())
+        case _ =>
+          false
+      }
+
+    override def remove(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          val thatValue = value.getValue()
+          if (self.containsKey(thatKey) && same(self.get(thatKey), thatValue))
+              same(self.remove(thatKey), thatValue)
+          else
+            false
+        case _ =>
+          false
+      }
+
+    override def clear(): Unit =
+      inner.clear()
+  }
+}
+
+object IdentityHashMap {
+  private final case class IdentityBox[+K](inner: K) {
+    override def equals(o: Any): Boolean = {
+      o match {
+        case o: IdentityBox[_] =>
+          // inner.asInstanceOf[AnyRef] eq o.inner.asInstanceOf[AnyRef]
+          // Use the following until 1.0.0
+          // See https://github.com/scala-js/scala-js/pull/3747
```

Comment:
This comment would apply to all the places where `same()` is called, which is quite a lot now. Consider moving the comment to the definition of `same` instead.

Reply:
Makes sense too because `same` is used in multiple places so we intend to keep that method.